app/models/robots/tube_rack_wrapper.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+module Robots
+  # This wrapper class is for tube racks that are not actual recorded labware.
+  # The instances acts as a labware wrapper and provides access to the tubes.
+  # Labware methods are delegated to the first tube on the tube rack.
+  #
+  class TubeRackWrapper
+    attr_accessor :barcode, :parent, :tubes
+    delegate :purpose_name, :purpose, :state, :uuid, to: :last_tube, allow_nil: true
+
+    # Initializes a new instance of the class.
+    #
+    # @param [LabwareBarcode] barcode the barcode object of the tube rack
+    # @param [Plate] parent the parent plate
+    # @tubes [Array<Tube>] the tubes on the tube rack
+    #
+    def initialize(barcode, parent, tubes: [])
+      @barcode = barcode
+      @parent = parent
+      @tubes = []
+      @tube_positions = {} # Keep track of tube positions for push performance
```

Comment:
Describe structure here? i.e. position coordinate => index in `@tubes` array (is that right)?

Reply:
Updated code comment.